ui/src/app/stores/workflows/workflows.reducers.ts

```diff
@@ -16,27 +16,141 @@
+import {WorkflowJoinedModel} from "../../models/workflowJoined.model";
+import {WorkflowFormPartsModel} from "../../models/workflowFormParts.model";
 
 export interface State {
   projects: ProjectModel[],
-  workflows: WorkflowModel[]
-  loading: boolean
+  workflows: WorkflowModel[],
+  loading: boolean,
+  workflowAction: {
+    id: number,
+    mode: string,
+    loading: boolean,
+    workflow: WorkflowJoinedModel,
+    workflowData: {
+      details: {property: string, value: any}[],
+      sensor: {property: string, value: any}[],
+      jobs: {order: number, job: {property: string, value: any}[]}[]
+    }
+  },
+  workflowFormParts: WorkflowFormPartsModel
 }
 
 const initialState: State = {
   projects: [],
   workflows: [],
-  loading: true
+  loading: true,
+  workflowAction: {
+    id: undefined,
+    mode: undefined,
+    loading: true,
+    workflow: undefined,
+    workflowData: {
+      details: [],
+      sensor: [],
+      jobs: []
+    }
+  },
+  workflowFormParts: undefined
 };
 
 export function workflowsReducer(state: State = initialState, action: WorkflowsActions.WorkflowsActions) {
   switch (action.type) {
     case (WorkflowsActions.INITIALIZE_WORKFLOWS):
       return {...state, loading: true};
     case (WorkflowsActions.INITIALIZE_WORKFLOWS_SUCCESS):
-      return {...state, loading: false, projects: action.payload.projects, workflows: action.payload.workflows};
+      return {...state, loading: false, projects: action.payload.projects, workflows: action.payload.workflows, workflowFormParts: action.payload.workflowFormParts};
     case (WorkflowsActions.INITIALIZE_WORKFLOWS_FAILURE):
       return {...initialState, loading: false};
+
+    case (WorkflowsActions.STAR_WORKFLOW_INITIALIZATION):
+      return {...state, workflowAction: {
+          ...initialState.workflowAction, id: action.payload.id, mode: action.payload.mode, loading: true
+        }};
+    case (WorkflowsActions.SET_EMPTY_WORKFLOW):
+      return {...state, workflowAction: {
+          ...initialState.workflowAction, loading: false
+        }};
+    case (WorkflowsActions.LOAD_WORKFLOW_SUCCESS):
+      return {...state, workflowAction: {
+          ...state.workflowAction, workflow: action.payload.workflow, loading: false,
+          workflowData: {...state.workflowAction.workflowData,
+            details: action.payload.detailsData, sensor: action.payload.sensorData, jobs: action.payload.jobsData}
+        }};
+    case (WorkflowsActions.LOAD_WORKFLOW_FAILURE):
+      return {...state, workflowAction: {
+          ...initialState.workflowAction, loading: false
+        }};
+    case (WorkflowsActions.LOAD_WORKFLOW_FAILURE_INCORRECT_ID):
+      return {...state, workflowAction: {
+          ...initialState.workflowAction, loading: false
+        }};
+    case (WorkflowsActions.WORKFLOW_DETAILS_CHANGED):
+      const detailsData = [
+        ...state.workflowAction.workflowData.details.filter(item => item.property !== action.payload.property),
```

Comment:
No. state.workflowAction.workflowData.details is array and action.payload is one element so you have to do it otherwise it won't override but add an element to the array. You can end up with duplicates.